src/main/kotlin/uk/gov/dluhc/printapi/messaging/service/CertificateFailedToPrintEmailSender.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.dluhc.printapi.messaging.service
+
+import mu.KotlinLogging
+import org.springframework.stereotype.Component
+import uk.gov.dluhc.printapi.client.ElectoralRegistrationOfficeManagementApiClient
+import uk.gov.dluhc.printapi.database.entity.Certificate
+import uk.gov.dluhc.printapi.dto.SendCertificateFailedToPrintEmailRequest
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage
+
+private val logger = KotlinLogging.logger {}
+
+@Component
+class CertificateFailedToPrintEmailSender(
+    electoralRegistrationOfficeManagementApiClient: ElectoralRegistrationOfficeManagementApiClient,
+    private val emailService: EmailService,
+) : AbstractEmailSender(electoralRegistrationOfficeManagementApiClient) {
+
+    override fun testAndSend(printResponse: ProcessPrintResponseMessage, certificate: Certificate): Boolean {
+        if (printResponse.status != ProcessPrintResponseMessage.Status.FAILED) {
```

Comment:
My understanding is that this service method will only be called when the print response status has failed, so unsure why do we need to do these checks within these components/services. Same query for `CertificateNotDeliveredEmailSender` if loop check

Reply:
Yes, I've added a similar comment 👍

Reply:
👍  I've moved these checks up to PrintResponseProcessingService#processPrintResponse